Assets/Scripts/Health.cs

```diff
@@ -8,6 +8,10 @@ public class Health : MonoBehaviour {
 	public int maxHealth = 10;
 	private int currentHealth;
 
+    public float dropProbability; // number between 0.0 and 1.0
```

Comment:
@xxfast reviewing code, regardless of this code been in the health scrip or another script they cannot be uncoupled.
the spawning of the item has to be part of the object death, either the heath is responsible for object death meaning the spawn of the drop is a part of health, or the drop is responsible for object death, meaning this has to trigger it. either way the scipt that handles the objects death is the scrip that spawns the pickup object.